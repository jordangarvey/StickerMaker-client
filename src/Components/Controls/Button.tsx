import React, { SFC } from "react";
import { Text, TouchableOpacity, StyleSheet } from "react-native";


/**
 * Interface for the Button props
 */
interface IButtonProps {
	/** The current tab of the template choosing process */
	onPress(): void;
	/** The button text */
	text?: string;
}

/**
 * Component to render a Button
 */
const Button: SFC<IButtonProps> = (props) => (
	<TouchableOpacity onPress={props.onPress} style={styles.button}>
		<Text>{props.text || props.children}</Text>
	</TouchableOpacity>
);

/**
 * Styles for the Button
 */
const styles = StyleSheet.create({
	button: {
		borderWidth: 1,
		borderRadius: 12,
		color: "black",
		fontSize: 24,
		overflow: "hidden",
		padding: 12,
		textAlign: "center"
	}
});

export default Button;
