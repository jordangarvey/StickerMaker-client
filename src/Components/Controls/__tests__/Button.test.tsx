import React from "react";

import renderer from "react-test-renderer";

import Button from "../Button";


describe("The Button componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<Button
				onPress={mockFunction}
				text="Test"
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
