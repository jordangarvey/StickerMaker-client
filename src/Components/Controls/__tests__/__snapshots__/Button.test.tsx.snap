// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button componentâ€™s snapshot renders as expected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 12,
      "borderWidth": 1,
      "color": "black",
      "fontSize": 24,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 12,
      "textAlign": "center",
    }
  }
>
  <Text>
    Test
  </Text>
</View>
`;
